static/js/eventemitter.js

```diff
@@ -0,0 +1,20 @@
+class EventEmitter {
```

Comment:
any reason you're not using an [EventTarget](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/EventTarget)?

Reply:
Nope! I've just missed that it exists (I thought it was just an interface)!